general/src/pii/processor.rs

```diff
@@ -672,7 +633,7 @@ fn test_basic_stripping() {
             },
             "applications": {
                 "$string": ["@ip"],
-                "$object": ["remove_bad_headers"]
+                "$object.**": ["remove_bad_headers"]
```

Comment:
This might be wrong, or at least a breaking change

Reply:
@mitsuhiko @jan-auer I want to change redactPair from a rule that works on containers/databags to a rule that operates on the value that it is supposed to remove. This means that `redactPair` should no longer be applied to e.g. the headers dict, but all header values. This is more consistent with how other rules behave.

I expect that most PII configs will have `**` as application anyway.

Reply:
@mitsuhiko @untitaker I think we might even be able to get rid of redact pair now by using our new selectors. Generally, we probably don't need fully-blown regexes for keys. So, instead we can match keys with globs like this:

```json
"$request.body.**.*password*": "@remove"
```

WDYT?

Reply:
Discussed offline. Let's accept the breaking change and remove redact_pair in a follow-up PR.